packages/subgraph/src/mapping.ts

```diff
@@ -748,3 +759,119 @@ export function handleExercisedCommitments(
     handleExercisedCommitment(event);
   });
 }
+
+export function handleCollateralEscrowDeployed(
+  event: CollateralEscrowDeployed
+): void {
+  const bid: Bid = loadBidById(event.params._bidId.toString());
+  bid.collateralEscrow = event.params._collateralEscrow;
+  bid.save();
+}
+
+export function handleCollateralEscrowDeployeds(
+  events: CollateralEscrowDeployed[]
+): void {
+  events.forEach(event => {
+    handleCollateralEscrowDeployed(event);
+  })
+}
+
+export function handleCollateralCommitted(
+    event: CollateralCommitted
+): void {
+  const bid: Bid = loadBidById(event.params._bidId.toString());
+  // Load collateral by bidId and collateral address
+  const collateral = loadCollateral(
+      event.params._bidId.toString(),
+      event.params._collateralAddress
+    );
+  collateral.amount = event.params._amount;
+  collateral.tokenId = event.params._tokenId;
+  collateral.type = getTypeString(event.params._type);
+  collateral.collateralAddress = event.params._collateralAddress;
+  collateral.bid = bid.id;
+  collateral.status = 'Committed'
+  collateral.receiver = bid.borrowerAddress;
+  collateral.save();
+  let collateralCommitted = bid.collateral;
+  if (collateralCommitted) {
+    collateralCommitted.push(collateral.id);
+    bid.collateral = collateralCommitted;
+    bid.save();
```

Comment:
what is this doing? looks like the schema is deriving the `collateral` array on a bid by using the `Collateral` entity's reference to the bid ID - I would imagine that we shouldn't need to do this and it still work